src/librepgp/stream-armour.c

```diff
@@ -0,0 +1,987 @@
+                    (*llen)--;
+                }
+                return true;
+            }
+        }
+        clen += read;
+    } while (clen < len);
+
+    return false;
+}
+
+static int
+armour_read_padding(pgp_source_t *src)
+{
+    char                        st[64];
+    size_t                      stlen;
+    pgp_source_armored_param_t *param = src->param;
+
+    if (!armour_peek_line(param->readsrc, st, 12, &stlen)) {
+        return -1;
+    }
+
+    if ((stlen == 1) || (stlen == 2)) {
+        if ((st[0] != '=') || ((stlen == 2) && (st[1] != '='))) {
+            return -1;
+        }
+
+        src_skip(param->readsrc, stlen);
+        armour_skip_eol(param->readsrc);
+        return stlen;
+    } else if (stlen == 5) {
+        return 0;
+    }
+
+    return -1;
+}
+
+static bool
+armour_read_crc(pgp_source_t *src)
+{
+    uint8_t                     dec[4];
+    char                        crc[8];
+    size_t                      clen;
+    pgp_source_armored_param_t *param = src->param;
+
+    if (!armour_peek_line(param->readsrc, crc, sizeof(crc), &clen)) {
+        return false;
+    }
+
+    if ((clen == 5) && (crc[0] == '=')) {
+        for (int i = 0; i < 4; i++) {
+            if ((dec[i] = B64DEC[(int) crc[i + 1]]) >= 64) {
+                return false;
+            }
+        }
+
+        param->readcrc = (dec[0] << 18) | (dec[1] << 12) | (dec[2] << 6) | (dec[3]);
+        src_skip(param->readsrc, 5);
+        armour_skip_eol(param->readsrc);
+        return true;
+    }
+
+    return false;
+}
+
+static bool
+armour_read_trailer(pgp_source_t *src)
+{
+    char                        st[64];
+    char                        str[64];
+    size_t                      stlen;
+    ssize_t                     read;
+    pgp_source_armored_param_t *param = src->param;
+
+    stlen = strlen(param->armourhdr);
+    strncpy(st, "-----END", 8);
+    strncpy(st + 8, param->armourhdr + 5, stlen - 5);
+    strncpy(st + stlen + 3, "-----", 5);
+    stlen += 8;
+    read = src_peek(param->readsrc, str, stlen);
+    if ((read < stlen) || strncmp(str, st, stlen)) {
+        return false;
+    }
+    src_skip(param->readsrc, stlen);
+    return true;
+}
+
+static ssize_t
+armoured_src_read(pgp_source_t *src, void *buf, size_t len)
+{
+    pgp_source_armored_param_t *param = src->param;
+    uint8_t  b64buf[ARMOURED_BLOCK_SIZE];     /* input base64 data with spaces and so on */
+    uint8_t  decbuf[ARMOURED_BLOCK_SIZE + 4]; /* decoded 6-bit values */
+    uint8_t *bufptr = buf;                    /* for better readability below */
+    uint8_t *bptr, *bend;                     /* pointer to input data in b64buf */
+    uint8_t *dptr, *dend, *pend; /* pointers to decoded data in decbuf: working pointer, last
+                                    available byte, last byte to process */
+    uint8_t  bval;
+    uint32_t b24;
+    ssize_t  read;
+    ssize_t  left = len;
+    int      eqcount; /* number of '=' at the end of base64 stream */
```

Comment:
Can you initialize this to 0, my compiler believes it can be used uninitialized (seems potentially true).

```
stream-armour.c:382:19: error: ‘eqcount’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
```

Reply:
Sure, fixed. Actually it will not be used not initialized but it relies on flag eofb64, which seems not to be clear for compiler.